docs/egress.md

```diff
@@ -0,0 +1,297 @@
+  by source IP in external firewall, etc.
+
+- You want to limit outgoing external connections leave the cluster via certain
+  Nodes, for security controls, or due to the network topology restriction.
+
+This guide demonstrates how to configure `Egress` to achieve the above result.
+
+## Prerequisites
+
+Egress is introduced in v1.0 as an alpha feature. As with other alpha features,
+a feature gate `Egress` must be enabled on the antrea-controller and
+antrea-agent for the feature to work. The following options in the
+`antrea-config` ConfigMap need to be set:
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: antrea-config-dcfb6k2hkm
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    featureGates:
+      Egress: true
+  antrea-controller.conf: |
+    featureGates:
+      Egress: true
+```
+
+## The Egress resource
+
+A typical Egress resource example:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: Egress
+metadata:
+  name: egress-prod-web
+spec:
+  appliedTo:
+    namespaceSelector:
+      matchLabels:
+        env: prod
+    podSelector:
+      matchLabels:
+        role: web
+  egressIP: 10.10.0.8
+  externalIPPool: prod-external-ip-pool
+status:
+  egressNode: node01
+```
+
+### AppliedTo
+
+The `appliedTo` field specifies the grouping criteria of Pods to which the
+Egress applies to. Pods can be selected cluster-wide using `podSelector`. If set
+with a `namespaceSelector`, all Pods from Namespaces selected by the
+`namespaceSelector` will be selected. Specific Pods from specific Namespaces can
+be selected by providing both a `podSelector` and a `namespaceSelector`. Empty
+`appliedTo` selects nothing. The field is mandatory.
+
+### EgressIP
+
+The `egressIP` field specifies the egress (SNAT) IP the traffic from the
+selected Pods to the external network should use. **The IP must be reachable
+from all Nodes.** The IP can be specified when creating the Egress. Starting
+with Antrea v1.2, it can be allocated from an `ExternalIPPool` automatically.
+
+- If `egressIP` is not specified, `externalIPPool` must be specified. An IP will
+  be allocated from the pool by the antrea-controller. The IP will be assigned
+  to a Node selected by the `nodeSelector` of the `externalIPPool` automatically.
+- If both `egressIP` and `externalIPPool` are specified, the IP must be in the
+  range of the pool. Similarly, the IP will be assigned to a Node selected by
+  the `externalIPPool` automatically.
+- If only `egressIP` is specified, Antrea will not manage the assignment of the
+  IP and it must be assigned to an arbitrary interface of one Node manually.
+
+**Starting with Antrea v1.2, high availability is provided automatically when
+the `egressIP` is allocated from an `externalIPPool`**, i.e. when the
+`externalIPPool` is specified. If the Node hosting the `egressIP` fails, another
+Node will be elected as the new egress Node of this Egress. It will take over
```

Comment:
suggestion: will be elected (from among the remaining Nodes selected by the `nodeSelector` of the `externalIPPool`) as the ...

Reply:
done